views/js/runner/providerLoader.js

```diff
@@ -0,0 +1,159 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 Open Assessment Technologies SA ;
+ */
+
+/**
+ * Loads all the required providers and the underlying modules
+ * required by the test runner.
+ *
+ *
+ * @example
+ * providerLoader({
+ *    "runner": {
+ *      "id": "mock-runner",
+ *      "module": "taoTests/test/runner/mocks/mockRunnerProvider",
+ *      "bundle": "taoTests/test/runner/mocks/mockBundle.min",
+ *      "category": "testrunner"
+ *    },
+ *    "proxy": {
+ *      "id": "mock-proxy",
+ *      "module": "taoTests/test/runner/mocks/mockProxyProvider",
+ *      "bundle": "taoTests/test/runner/mocks/mockBundle.min",
+ *      "category": "online"
+ *    },
+ *    "communicator": {
+ *      "id": "request",
+ *      "module": "core/communicator/request",
+ *      "bundle": "loader/vendor.min",
+ *      "category": "request"
+ *    },
+ *    "plugins": [{
+ *      "id": "fooglin",
+ *      "module": "taoTests/test/runner/mocks/mockPlugin1",
+ *      "bundle": "taoTests/test/runner/mocks/mockBundle.min",
+ *      "category": "content"
+ *    }, {
+ *      "id": "barglin",
+ *      "module": "taoTests/test/runner/mocks/mockPlugin2",
+ *      "bundle": "taoTests/test/runner/mocks/mockBundle.min",
+ *      "category": "tools"
+ *    }]
+ *  }, false)
+ *  .then( ({ runnerProvider, plugins }) => {
+ *      //...
+ *  })
+ *  .catch( err => console.error(err) );
+ *
+ *
+ * @author Bertrand Chevrier <bertrand@taotesting.com>
+ */
+define([
+    'core/logger',
+    'core/providerLoader',
+    'core/pluginLoader',
+    'core/communicator',
+    'taoTests/runner/runner',
+    'taoTests/runner/proxy',
+], function(loggerFactory, providerLoader, pluginLoader, communicator, runner, proxy){
+    'use strict';
+
+    const logger = loggerFactory('taoTests/runner/loader');
+
+    /**
+     * Load the providers that match the registration
+     * @param {Object} providers
+     * @param {Object|Object[]} providers.runner
```

Comment:
It could be useful to have a description of what should be an expected provider here. That would also prevent to think we could give the providers either as an array or as a key/value. For instance:
```javascript
/**
 * @typedef {Object} provider - A provider is an object exposing a list of methods with respect to the API managed by the target.
 * @property {String} name - The name of the provider. It should be unique among all.
 * @property {Function} init - Each provider much expose at least a method `init()`
 * @property {Function} ... - Any other method the target is expecting
 */
...
/**
 * @param {provider|provider[]} providers.runner
 * @param {provider|provider[]} providers.proxy
 */
...